### ISSUE TITLE
requestAnimationFrame being called in an infinite loop when truncate elements not visible on the page

### ISSUE THREAD

**Issue by jordie23** [One-com/react-truncate#61]
We have a scenario where the Truncate component will potentially not be visible, and this is leading to `requestAnimationFrame` being called continuously. 

In https://github.com/One-com/react-truncate/blob/master/src/Truncate.js#L133 on line 133 and following, it checks the width of the parent and if the width is zero, it will re-queue another check with `requestAnimationFrame`. As the parent or a grandparent is invisible (display: none) in our scenario, the width will always be zero, resulting in this being calling indefinitely. 

I think a check before might solve this. The generally accepted way of determining if an element is visible on the page or not is to use `element.offsetHeight`. 

```
const targetParentVisible = target.parentNode.offsetHeight;

// If target isn't visible on the page, then it'll start an infinite
// loop with requestAnimationFrame below
if(!targetParentVisible) {
  return
}
```

**Comment by jordie23** [comment]
PR https://github.com/One-com/react-truncate/pull/67

**Comment by PabloSichert** [comment]
I think we could run into a problem here - for the first render the parent node is never present, since the mount order of react is `Child -> Parent`.

If we simply return the function at this point, the recalculation never happens.

**Comment by jordie23** [comment]
While it was working fine for me, it's possible it was just good timing.

I've modified it to check the target element instead of its parent: https://github.com/One-com/react-truncate/pull/67

**Comment by PabloSichert** [comment]
I agree that the infinite loop is all but optimal - but how do you expect the component to behave once the parent gets visible?

We can either:
- not calculate the width, but fail to "wake it up" once it gets visible
- calculate the width indefinitely until it gets visible

This seems to be a pitfall no matter how you twist it.

---

By the way - did you ever profile this case? The loop is inside a `requestAnimationFrame`, so the overhead of doing this might be less than one would assume at first.

I think it would be nice to look at some data before continuing here.

In any case - thanks for pointing out this issue. I'd leave a note on the [known issues](https://github.com/One-com/react-truncate#known-issues) section to inform the users of this library of the consequences.
